sherpa/astro/flux.py

```diff
@@ -529,12 +529,91 @@ def sample_flux(fit, data, src,
 
 def calc_sample_flux(id, lo, hi, session, fit, data, samples, modelcomponent,
                      confidence):
+    """Given a set of parameter samples, estimate the flux distribution.
+
+    This is similar to sample_flux but returns values for both the full
+    and component models (which can be the same). The set of parameter
+    samples is searched to remove rows where the parameters lie outside
+    the soft limits, and then this set of rows is used to calculate the
+    flux of the modelcomponent expression (the flux for the full model
+    is taken from the samples argument). The resulting set of fluxes is
+    used to calculate the median and the confidence range.
+
+    .. versionchanged:: 4.13.1
+       The clipping now includes parameters at the soft limits;
+       previously they were excluded which could cause problems with
+       parameters for which we can only calculate an upper limit.
+
+    Parameters
+    ----------
+    id : int or str
+        The dataset identifier. It must exist and have an associated model in
+        session.
```

Comment:
```suggestion
        the session.
```

Reply:
I am reluctant to fix this as I plan to remove this parameter (and a bunch of other ones) in a forthcoming PR...

Reply:
Well, actually my real hope is to be able to remove this function completely...